core/utils/metrics.js

```diff
@@ -0,0 +1,124 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview Workspace metrics definitions.
+ * @author samelh@google.com (Sam El-Husseini)
+ */
+'use strict';
+
+goog.provide('Blockly.utils.Metrics');
+
+
+/**
+ * @record
+ */
+Blockly.utils.Metrics = function() {};
+
+/**
+ * Height of the visible portion of the workspace.
+ * @type {number}
+ */
+Blockly.utils.Metrics.prototype.viewHeight;
+
+/**
+ * Width of the visible portion of the workspace.
+ * @type {number}
+ */
+Blockly.utils.Metrics.prototype.viewWidth;
+
+/**
+ * Height of the content.
+ * @type {number}
+ */
+Blockly.utils.Metrics.prototype.contentHeight;
+
+/**
+ * Width of the content.
+ * @type {number}
+ */
+Blockly.utils.Metrics.prototype.contentWidth;
+
+/**
+ * Top-edge of the visible portion of the workspace, relative to the workspace
+ * origin.
+ * @type {number}
+ */
+Blockly.utils.Metrics.prototype.viewTop;
+
+/**
+ * Left-edge of the visible portion of the workspace, relative to the workspace
+ * origin.
+ * @type {number}
+ */
+Blockly.utils.Metrics.prototype.viewLeft;
+
+/**
+ * Top-edge of the content, relative to the workspace origin.
+ * @type {number}
+ */
+Blockly.utils.Metrics.prototype.contentTop;
+
+/**
+ * Left-edge of the content relative to the workspace origin.
+ * @type {number}
+ */
+Blockly.utils.Metrics.prototype.contentLeft;
+
+/**
+ * Top-edge of the visible portion of the workspace, relative to the blocklyDiv.
+ * @type {number}
+ */
+Blockly.utils.Metrics.prototype.absoluteTop;
+
+/**
+ * Left-edge of the visible portion of the workspace, relative to the
+ * blocklyDiv.
+ * @type {number}
+ */
+Blockly.utils.Metrics.prototype.absoluteLeft;
+
+/**
+ * Height of the Blockly div (the view + the toolbox, simple of otherwise).
+ * @type {number|undefined}
+ */
+Blockly.utils.Metrics.prototype.svgHeight;
+
+/**
+ * Width of the Blockly div (the view + the toolbox, simple or otherwise).
+ * @type {number|undefined}
+ */
+Blockly.utils.Metrics.prototype.svgWidth;
+
+/**
+ * Width of the toolbox, if it exists.  Otherwise zero.
+ * @type {number|undefined}
+ */
+Blockly.utils.Metrics.prototype.toolboxWidth;
+
+/**
+ * Height of the toolbox, if it exists.  Otherwise zero.
+ * @type {number|undefined}
+ */
+Blockly.utils.Metrics.prototype.toolboxHeight;
+
+/**
+ * Width of the flyout if it is always open.  Otherwise zero.
+ * @type {number|undefined}
+ */
+Blockly.utils.Metrics.prototype.flyoutWidth;
+
+/**
+ * Height of the flyout if it is always open.  Otherwise zero.
+ * @type {number|undefined}
+ */
+Blockly.utils.Metrics.prototype.flyoutHeight;
+
+/**
+ * Top, bottom, left or right. Use TOOLBOX_AT constants to compare.
+ * @type {number|undefined}
+ */
+Blockly.utils.Metrics.prototype.toolboxPosition;
```

Comment:
nit: move this up to be with the toolbox group

Reply:
I wanted to keep the ones that could be undefined separately and ordered as they are ordered in getMetrics, ie: conceptually.
I could be order them alphabetically into two groups if you feel that would help?

Reply:
Yeah that could help, I just want them to be ordered in some way that I understand where to put something new when looking at it.

Reply:
Totally, they are currently grouped as they are in the "main" getMetrics method https://github.com/google/blockly/blob/develop/core/workspace_svg.js#L2425. I don't like alphabetical grouping here because its nice having width and height of the same category coupled together. 

Happy to bump toolboxPosition so that it's closer to toolboxWidth and toolboxHeight if that makes more sense to you, but I would say we should apply the same order in `Blockly.WorkspaceSvg.getTopLevelWorkspaceMetrics_`.

Reply:
Also, updated ordering of metrics to match everywhere.